### PR TITLE
Update index page introduction

### LINKED ISSUE #730: Rewrite introduction paragraph in documentation
I just had a coworker ask if I had a paragraph describing Satpy that he could include in a proposal for a project that will use Satpy. I pointed him to the documentation but quickly realized that the first paragraph of the sphinx docs home page is not a very good summary/description of Satpy as a whole. We/I should rewrite this. This is what it is currently:

> Satpy is a python library for reading and manipulating meteorological remote sensing data and writing it to various image and data file formats. Satpy comes with the ability to make various RGB composites directly from satellite instrument channel data or higher level processing output. The pyresample package is used to resample data to different uniform areas or grids. Various atmospheric corrections and visual enhancements are also provided, either directly in Satpy or from those in the PySpectral and TrollImage packages.

### PR DESCRIPTION
**Pull request by djhoese**
As mentioned in #730 I felt that the introduction paragraph wasn't very helpful to new users or as a good description of its features.

 - [x] Closes #730 
 - [x] Fully documented

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -2,19 +2,33 @@
...
-to resample data to different uniform areas or grids. Various atmospheric
-corrections and visual enhancements are also provided, either directly in
-Satpy or from those in the
-`PySpectral <https://pyspectral.readthedocs.io/en/develop/>`_ and
-`TrollImage <http://trollimage.readthedocs.io/en/latest/>`_ packages.
-
-Go to the project_ page for source code and downloads.
+Satpy is a python library for reading, manipulating, and writing data from
+remote-sensing earth-observing satellite instruments. Satpy provides users
+with a series of readers that convert satellite instrument file formats to
```
Comment (sfinkens) [inline doc/source/index.rst]: What about a little physics here: "Satpy provides users with a series of readers that convert raw satellite data to physical quantities such as radiance, reflectance or brightness temperature (so called calibration). The calibrated data is returned in the common Xarray..."

Comment (djhoese) [reply doc/source/index.rst]: We have readers for other types of data (cloud masks, fires, other types of temperatures, cloud heights, etc). Does telling users about radiance/reflectance/BTs add anything to the description? I'm worried that it hurts the "executive summary" nature of the paragraph...but then again I mention Xarray objects.

Comment (sfinkens) [reply doc/source/index.rst]: Good point, I was too narrow-minded there. What about "Satpy provides users with a series of readers that convert a multitude of geophysical parameters in different file formats to the common Xarray DataArray and Dataset classes". I think mentioning Xarray is a good thing.

Comment (sfinkens) [reply doc/source/index.rst]: Maybe "satellite-derived geophysical parameters"

Comment (sfinkens) [reply doc/source/index.rst]: Or satellite-borne?

Comment (mraspaud) [reply doc/source/index.rst]: I like @sfinkens 's proposal

**Pull request merged**